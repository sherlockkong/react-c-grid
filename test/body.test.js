import React from 'react';
import Body from '../src/components/Body';
import renderer from 'react-test-renderer';

it('body', () => {
	const columns = [
		{ key: 'name', label: 'Name' },
		{ key: 'email', label: 'Email' },
		{ key: 'company', label: 'Company' },
		{ key: 'phone', label: 'Phone' },
		{ key: 'city', label: 'City' },
		{ key: 'country', label: 'Country' },
		{ key: 'postal', label: 'Postal' }
	];
	const rows = [
		{
			"name": "Stewart",
			"email": "nec@auctorMaurisvel.edu",
			"company": "Amet Diam Eu PC",
			"phone": "1675071109899",
			"city": "San Marcello",
			"country": "Iceland",
			"postal": "14778"
		}
	];

	const props = { rows, columns };

	const body = renderer
		.create(<Body {...props} />)
		.toJSON();
	expect(body).toMatchSnapshot();
});

it('body change row height', () => {
	const columns = [
		{ key: 'name', label: 'Name' },
		{ key: 'email', label: 'Email' },
		{ key: 'company', label: 'Company' },
		{ key: 'phone', label: 'Phone' },
		{ key: 'city', label: 'City' },
		{ key: 'country', label: 'Country' },
		{ key: 'postal', label: 'Postal' }
	];
	const rows = [
		{
			"name": "Stewart",
			"email": "nec@auctorMaurisvel.edu",
			"company": "Amet Diam Eu PC",
			"phone": "1675071109899",
			"city": "San Marcello",
			"country": "Iceland",
			"postal": "14778"
		}
	];

	const props = { rows, columns, rowHeight: 66 };

	const body = renderer
		.create(<Body {...props} />)
		.toJSON();
	expect(body).toMatchSnapshot();
});

it('body custom render cell', () => {
	const columns = [
		{ key: 'name', label: 'Name' },
		{ key: 'email', label: 'Email' },
		{ key: 'company', label: 'Company' },
		{ key: 'phone', label: 'Phone' },
		{ key: 'city', label: 'City' },
		{ key: 'country', label: 'Country' },
		{ key: 'postal', label: 'Postal' }
	];
	const rows = [
		{
			"name": "Stewart",
			"email": "nec@auctorMaurisvel.edu",
			"company": "Amet Diam Eu PC",
			"phone": "1675071109899",
			"city": "San Marcello",
			"country": "Iceland",
			"postal": "14778"
		}
	];

	const props = { rows, columns, rowHeight: 66, onRenderCell: (key, row) => `custom-${row[key]}` };

	const body = renderer
		.create(<Body {...props} />)
		.toJSON();
	expect(body).toMatchSnapshot();
});