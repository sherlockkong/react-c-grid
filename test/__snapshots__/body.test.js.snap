// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body 1`] = `
<div
  className="cg-body"
>
  <div
    className="place-holder-row"
    style={
      Object {
        "height": "48px",
        "lineHeight": "48px",
      }
    }
  />
  <div
    className="cg-row odd"
    style={
      Object {
        "height": "48px",
        "lineHeight": "48px",
      }
    }
  >
    <div
      className="cg-col-0 cg-cell"
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
        }
      }
    >
      Stewart
    </div>
    <div
      className="cg-col-1 cg-cell"
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
        }
      }
    >
      nec@auctorMaurisvel.edu
    </div>
    <div
      className="cg-col-2 cg-cell"
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
        }
      }
    >
      Amet Diam Eu PC
    </div>
    <div
      className="cg-col-3 cg-cell"
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
        }
      }
    >
      1675071109899
    </div>
    <div
      className="cg-col-4 cg-cell"
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
        }
      }
    >
      San Marcello
    </div>
    <div
      className="cg-col-5 cg-cell"
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
        }
      }
    >
      Iceland
    </div>
    <div
      className="cg-col-6 cg-cell"
      style={
        Object {
          "height": "48px",
          "lineHeight": "48px",
        }
      }
    >
      14778
    </div>
  </div>
</div>
`;

exports[`body change row height 1`] = `
<div
  className="cg-body"
>
  <div
    className="place-holder-row"
    style={
      Object {
        "height": "66px",
        "lineHeight": "66px",
      }
    }
  />
  <div
    className="cg-row odd"
    style={
      Object {
        "height": "66px",
        "lineHeight": "66px",
      }
    }
  >
    <div
      className="cg-col-0 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      Stewart
    </div>
    <div
      className="cg-col-1 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      nec@auctorMaurisvel.edu
    </div>
    <div
      className="cg-col-2 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      Amet Diam Eu PC
    </div>
    <div
      className="cg-col-3 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      1675071109899
    </div>
    <div
      className="cg-col-4 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      San Marcello
    </div>
    <div
      className="cg-col-5 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      Iceland
    </div>
    <div
      className="cg-col-6 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      14778
    </div>
  </div>
</div>
`;

exports[`body custom render cell 1`] = `
<div
  className="cg-body"
>
  <div
    className="place-holder-row"
    style={
      Object {
        "height": "66px",
        "lineHeight": "66px",
      }
    }
  />
  <div
    className="cg-row odd"
    style={
      Object {
        "height": "66px",
        "lineHeight": "66px",
      }
    }
  >
    <div
      className="cg-col-0 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      custom-Stewart
    </div>
    <div
      className="cg-col-1 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      custom-nec@auctorMaurisvel.edu
    </div>
    <div
      className="cg-col-2 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      custom-Amet Diam Eu PC
    </div>
    <div
      className="cg-col-3 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      custom-1675071109899
    </div>
    <div
      className="cg-col-4 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      custom-San Marcello
    </div>
    <div
      className="cg-col-5 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      custom-Iceland
    </div>
    <div
      className="cg-col-6 cg-cell"
      style={
        Object {
          "height": "66px",
          "lineHeight": "66px",
        }
      }
    >
      custom-14778
    </div>
  </div>
</div>
`;
